src/editor/nodes/ParticleNode.js

```diff
@@ -0,0 +1,416 @@
+          colorFactor[i] = this.Even(colorFactor[i]);
+          break;
+        case "Ease-in":
+          colorFactor[i] = this.EaseIn(colorFactor[i]);
+          break;
+        case "Ease-out":
+          colorFactor[i] = this.EaseOut(colorFactor[i]);
+          break;
+        case "Ease-in, out":
+          colorFactor[i] = this.EaseInOut(colorFactor[i]);
+          break;
+      }
+
+      if (colorFactor[i] <= 0.5) {
+        color[i * 4] = this.lerp(this.startColor.r, this.middleColor.r, colorFactor[i] / 0.5);
+        color[i * 4 + 1] = this.lerp(this.startColor.g, this.middleColor.g, colorFactor[i] / 0.5);
+        color[i * 4 + 2] = this.lerp(this.startColor.b, this.middleColor.b, colorFactor[i] / 0.5);
+        color[i * 4 + 3] = this.lerp(this.startOpacity, this.middleOpacity, colorFactor[i] / 0.5);
+      } else if (colorFactor[i] > 0.5) {
+        color[i * 4] = this.lerp(this.middleColor.r, this.endColor.r, (colorFactor[i] - 0.5) / 0.5);
+        color[i * 4 + 1] = this.lerp(this.middleColor.g, this.endColor.g, (colorFactor[i] - 0.5) / 0.5);
+        color[i * 4 + 2] = this.lerp(this.middleColor.b, this.endColor.b, (colorFactor[i] - 0.5) / 0.5);
+        color[i * 4 + 3] = this.lerp(this.middleOpacity, this.endOpacity, (colorFactor[i] - 0.5) / 0.5);
+      }
+
+      this.geometry.attributes.position.needsUpdate = true;
+      this.geometry.attributes.color.needsUpdate = true;
+      this.geometry.attributes.customAngle.needsUpdate = true;
+    }
+  }
+
+  lerp(start, end, a) {
+    return (end - start) * a + start;
+  }
+
+  Even(k) {
+    return k * k;
+  }
+
+  EaseIn(k) {
+    return k * k * k * k; //Quadratic
+  }
+
+  EaseOut(k) {
+    return Math.sin((k * Math.PI) / 2);
+  }
+
+  EaseInOut(k) {
+    if ((k *= 2) < 1) {
+      return 0.5 * k * k * k * k * k;
+    }
+
+    return 0.5 * ((k -= 2) * k * k * k * k + 2);
+  }
+
+  smoothstep(min, max, x) {
+    x = this.clamp((x - min) / (max - min), 0.0, 1.0);
+    return x * x * (3 - 2 * x);
+  }
+
+  clamp(min, max, x) {
+    if (x < min) x = min;
+    if (x > max) x = max;
+    return x;
+  }
+
+  serialize() {
+    return super.serialize({
+      particle: {
+        src: this._canonicalUrl,
+        emitterHeight: this.emitterHeight,
+        emitterWidth: this.emitterWidth,
+        startColor: this.startColor,
+        middleColor: this.middleColor,
+        endColor: this.endColor,
+        startOpacity: this.startOpacity,
+        middleOpacity: this.middleOpacity,
+        endOpacity: this.endOpacity,
+        size: this.size,
+        velocity: this.velocity,
+        endVelocity: this.endVelocity,
+        angularVelocity: this.angularVelocity,
+        particleCount: this.particleCount,
+        lifetime: this.lifetime,
+        lifetimeRandomness: this.lifetimeRandomness,
+        colorCurve: this.colorCurve
+      }
+    });
+  }
+
+  prepareForExport() {
+    super.prepareForExport();
+    this.addGLTFComponent("particle", {
+      src: this._canonicalUrl
```

Comment:
All of the properties that you are serializing above can be added to the glTF component so we can use that data in Hubs.